include/aspect/compat.h

```diff
@@ -65,6 +65,347 @@ namespace aspect
       }
   };
 }
+
+#include <deal.II/grid/tria_boundary_lib.h>
```

Comment:
I thought for compat.h we always included headers at the position of the code that uses them to remember to remove them. When they are at the top they might stay around. Is that not correct?